src/server/controllers/match.controller.js

```diff
@@ -49,15 +51,20 @@ export function getMentorList(req, res, next) {
   Promise.all([
     findConnection(matchOptions.option1, projectOption, localField.mentee),
     findConnection(matchOptions.option2, projectOption, localField.mentor),
-    findConnection(matchOptions.option3, projectOption, 'mentee_id'),
+    findConnection(matchOptions.option3, projectOption, localField.mentee),
   ])
     .then((results) => {
-      let exceptionList = [];
-
       results[0].forEach(user => exceptionList.push(user.mentee_id));
       results[1].forEach(user => exceptionList.push(user.mentor_id));
       results[2].forEach(user => pendingList.push(user.mentor_id.toString()));
 
+      return User.findOne({ _id: req.user._id }).exec();
```

Comment:
I added promise to Promise.all and used result[3].
But result[3] things are not so nice to recognize for code.

I don't know which way is the best to make code neat, but agree they are all independent promises that don't need to be asynchronized.